rustls/src/msgs/handshake.rs

```diff
@@ -865,7 +861,7 @@ impl Codec for ClientHelloPayload {
         let mut ret = ClientHelloPayload {
             client_version: ProtocolVersion::read(r)?,
             random: Random::read(r)?,
-            session_id: SessionID::read(r)?,
+            session_id: Some(SessionID::read(r)?),
```

Comment:
All instances of `Some(SessionID::read(r)?)` can be replaced with just `SessionID::read(r)`.